src/Func/func_platform.h

```diff
@@ -43,7 +43,7 @@ using WarblePtr = std::shared_ptr<WarbleServiceAbstraction>;
 using EventType = unsigned int;
 using FunctionName = std::string;
 using FunctionType =
-    std::function<PayloadOptional(WarbleServiceAbstraction &, Payload)>;
+    std::function<PayloadOptional(WarbleServiceAbstraction &, Payload, StoragePtr)>;
```

Comment:
Does moving Storage pointer to the func map affect the Ephemeral feature?

Reply:
No. Actually, either storing the storage pointer as a member function or passing it as parameters of functions meets the requirement of ephemeral, which means warble class has no states.
However,  "directly passing it to the functions will make the function signatures for those functions a bit longer, but it will make it clearer that they rely upon no state" quoted from Professor.

Reply:
Sounds reasonable!